service/history/historyEngine.go

```diff
@@ -581,6 +581,7 @@ func (e *historyEngineImpl) getMutableState(ctx ctx.Context,
 
 	executionInfo := msBuilder.GetExecutionInfo()
 	execution.RunId = context.getExecution().RunId
+	versionHistories := msBuilder.GetAllBranches()
```

Comment:
could you also add a TODO in GetMutableState function (the one above), specifying that we need to take branching into consideration?